crates/net/nat/src/lib.rs

```diff
@@ -21,7 +22,7 @@ use std::{
 use tracing::warn;
 
 /// All builtin resolvers.
-#[derive(Debug, Clone, Copy, Eq, PartialEq, Default, Hash)]
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Default, Hash, Serialize, Deserialize)]
```

Comment:
I think these should be manually implemented with `FromStr` and `Display`

Reply:
I will use SerializeDisplay and DeserializeFromStr from serde_with if that's ok?